pkg/apis/controlplane/v1beta1/conversion.go

```diff
@@ -124,13 +126,20 @@ func Convert_v1beta1_GroupMemberPod_To_controlplane_GroupMember(in *GroupMemberP
 	return nil
 }
 
-func Convert_controlplane_GroupMember_To_v1beta1_GroupMemberPod(in *controlplane.GroupMember, out *GroupMemberPod, s conversion.Scope) error {
+// Convert_controlplane_GroupMember_To_v1beta1_GroupMemberPod converts controlplane GroupMember to v1beta1 GroupMember
+// based on whether it's required to include Pod reference in the result. We must not include Pod reference when the
+// conversion is called for an AddressGroup as agents don't expect it in v1beta1 version.
+// This function doesn't match the pattern of conversion function which requires the last parameter to be
+// conversion.Scope so won't be registered to schema.
+func Convert_controlplane_GroupMember_To_v1beta1_GroupMemberPod(in *controlplane.GroupMember, out *GroupMemberPod, includePodRef bool) error {
```

Comment:
could have been a private function?

Reply:
Definitely can be private, I kept its name as it was as code in this file doesn't follow normal go style anyway.
Do you prefer it to be private?